datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1446,18 +1446,54 @@ impl<'a, S: SimplifyInfo> TreeNodeRewriter for Simplifier<'a, S> {
 
             // Rules for Like
             Expr::Like(Like {
-                expr,
-                pattern,
+                expr: ref like_expr,
```

Comment:
keep `expr` as `expr` 
use `like_expr` when referring to the whole `Expr::Like` thing